pkgs/applications/science/logic/monosat/default.nix

```diff
@@ -20,10 +20,12 @@ let
     inherit rev sha256;
   };
 
+  # The CMake build is set up to build both shared and static libraries, so
+  # both shared and static libraries are needed as dependencies.
   core = stdenv.mkDerivation rec {
     name = "${pname}-${version}";
     inherit src;
-    buildInputs = [ cmake zlib gmp jdk8 ];
+    buildInputs = [ cmake zlib zlib.static gmp-static jdk8 git ];
```

Comment:
It looks like you've removed the dynamic gmp dependency. Does the dynamic library still get built?

Reply:
Unlike zlib, which has its dynamic and static libraries separated into the `zlib` and `zlib.static` derivations, gmp seems to have a dynamic derivation and a dynamic+static derivation (built by setting the `withStatic` argument to true), so yes the dynamic library does get built.